capstone-frontend/src/components/ListPage.jsx

```diff
@@ -34,34 +37,59 @@ class ListPage extends Component {
       gBooks.push(gBook[0]);
     }))
 
-    this.setState({ gBooks, loading: false });
+    this.setState({ gBooks, bookList: bookList[0], loading: false });
   }
 
-  componentDidMount() {
-    this.fetchBooks();
+  async componentDidMount() {
+    await this.fetchBooks();
   }
 
-  componentDidUpdate(prevProps) {
+  async componentDidUpdate(prevProps) {
     if (this.props.match.params.id !== prevProps.match.params.id) {
-      this.setState({ loading: true, empty: false })
-      this.fetchBooks();
+      this.setState({ loading: true, gBooks: [], bookList: {} })
+      await this.fetchBooks();
     }
   }
 
-  deleteBook = (bookId) => {
+  updateListPage = async () => {
+    // Restart 
+    this.setState(
+      {
+        loading: true,
+        bookList: {},
+        gBooks: [],
+      }
+    );
+
+    await this.fetchBooks();
+  }
+
+
+  deleteBook = async (bookId) => {
+
+    const gBooks = this.state.gBooks.filter(gBook => gBook.id !== bookId);
```

Comment:
What exactly does this do?

Reply:
Filter return a new array reference, which allows react to correctly update the array.